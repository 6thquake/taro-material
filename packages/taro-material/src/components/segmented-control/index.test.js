import Nerv from 'nervjs'
import { renderToString } from 'nerv-server'

import AtSegmentedControl from '../../../.temp/components/segmented-control/index'

describe('AtSegmentedControl Snap', () => {
  const values = ['tab1', 'tab2', 'tab3']

  it('render initial AtSegmentedControl', () => {
    const componet = renderToString(<AtSegmentedControl isTest />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtSegmentedControl -- props customStyle', () => {
    const componet = renderToString(<AtSegmentedControl isTest customStyle='color:red;' />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtSegmentedControl -- props className', () => {
    const componet = renderToString(<AtSegmentedControl isTest className='test' />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtSegmentedControl -- props values', () => {
    const componet = renderToString(<AtSegmentedControl isTest values={values} />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtSegmentedControl -- props current', () => {
    const componet = renderToString(<AtSegmentedControl isTest current={2} values={values} />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtSegmentedControl -- props color', () => {
    const componet = renderToString(<AtSegmentedControl isTest color='red' values={values} />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtSegmentedControl -- props fontSize', () => {
    const componet = renderToString(<AtSegmentedControl isTest fontSize='30' values={values} />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtSegmentedControl -- props disabled', () => {
    const componet = renderToString(<AtSegmentedControl isTest disabled values={values} />)
    expect(componet).toMatchSnapshot()
  })

  it('render AtSegmentedControl -- props selectedColor', () => {
    const componet = renderToString(<AtSegmentedControl isTest selectedColor='#fff' values={values} />)
    expect(componet).toMatchSnapshot()
  })
})
