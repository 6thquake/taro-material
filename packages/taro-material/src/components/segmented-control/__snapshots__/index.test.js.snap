// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtSegmentedControl Snap render AtSegmentedControl -- props className 1`] = `"<div style=\\"border-color: #6190E8;\\" class=\\"at-segmented-control test\\"></div>"`;

exports[`AtSegmentedControl Snap render AtSegmentedControl -- props color 1`] = `
"<div style=\\"border-color: #6190E8;\\" class=\\"at-segmented-control\\"><div style=\\"
      color: red;
      background-color:#6190E8;
      border-color: #6190E8;
      font-size: 0.5974rem;
    \\" class=\\"at-segmented-control__item\\">tab1</div><div style=\\"
      color: #6190E8;
      background-color:red;
      border-color: #6190E8;
      font-size: 0.5974rem;
    \\" class=\\"at-segmented-control__item\\">tab2</div><div style=\\"
      color: #6190E8;
      background-color:red;
      border-color: #6190E8;
      font-size: 0.5974rem;
    \\" class=\\"at-segmented-control__item\\">tab3</div></div>"
`;

exports[`AtSegmentedControl Snap render AtSegmentedControl -- props current 1`] = `
"<div style=\\"border-color: #6190E8;\\" class=\\"at-segmented-control\\"><div style=\\"
      color: #6190E8;
      background-color:#fff;
      border-color: #6190E8;
      font-size: 0.5974rem;
    \\" class=\\"at-segmented-control__item\\">tab1</div><div style=\\"
      color: #6190E8;
      background-color:#fff;
      border-color: #6190E8;
      font-size: 0.5974rem;
    \\" class=\\"at-segmented-control__item\\">tab2</div><div style=\\"
      color: #fff;
      background-color:#6190E8;
      border-color: #6190E8;
      font-size: 0.5974rem;
    \\" class=\\"at-segmented-control__item\\">tab3</div></div>"
`;

exports[`AtSegmentedControl Snap render AtSegmentedControl -- props customStyle 1`] = `"<div style=\\"border-color: #6190E8;color:red;\\" class=\\"at-segmented-control\\"></div>"`;

exports[`AtSegmentedControl Snap render AtSegmentedControl -- props disabled 1`] = `
"<div style=\\"border-color: #6190E8;\\" class=\\"at-segmented-control at-segmented-control--disabled\\"><div style=\\"
      color: #fff;
      background-color:#6190E8;
      border-color: #6190E8;
      font-size: 0.5974rem;
    \\" class=\\"at-segmented-control__item\\">tab1</div><div style=\\"
      color: #6190E8;
      background-color:#fff;
      border-color: #6190E8;
      font-size: 0.5974rem;
    \\" class=\\"at-segmented-control__item\\">tab2</div><div style=\\"
      color: #6190E8;
      background-color:#fff;
      border-color: #6190E8;
      font-size: 0.5974rem;
    \\" class=\\"at-segmented-control__item\\">tab3</div></div>"
`;

exports[`AtSegmentedControl Snap render AtSegmentedControl -- props fontSize 1`] = `
"<div style=\\"border-color: #6190E8;\\" class=\\"at-segmented-control\\"><div style=\\"
      color: #fff;
      background-color:#6190E8;
      border-color: #6190E8;
      font-size: 0.64rem;
    \\" class=\\"at-segmented-control__item\\">tab1</div><div style=\\"
      color: #6190E8;
      background-color:#fff;
      border-color: #6190E8;
      font-size: 0.64rem;
    \\" class=\\"at-segmented-control__item\\">tab2</div><div style=\\"
      color: #6190E8;
      background-color:#fff;
      border-color: #6190E8;
      font-size: 0.64rem;
    \\" class=\\"at-segmented-control__item\\">tab3</div></div>"
`;

exports[`AtSegmentedControl Snap render AtSegmentedControl -- props selectedColor 1`] = `
"<div style=\\"border-color: #fff;\\" class=\\"at-segmented-control\\"><div style=\\"
      color: #fff;
      background-color:#fff;
      border-color: #fff;
      font-size: 0.5974rem;
    \\" class=\\"at-segmented-control__item\\">tab1</div><div style=\\"
      color: #fff;
      background-color:#fff;
      border-color: #fff;
      font-size: 0.5974rem;
    \\" class=\\"at-segmented-control__item\\">tab2</div><div style=\\"
      color: #fff;
      background-color:#fff;
      border-color: #fff;
      font-size: 0.5974rem;
    \\" class=\\"at-segmented-control__item\\">tab3</div></div>"
`;

exports[`AtSegmentedControl Snap render AtSegmentedControl -- props values 1`] = `
"<div style=\\"border-color: #6190E8;\\" class=\\"at-segmented-control\\"><div style=\\"
      color: #fff;
      background-color:#6190E8;
      border-color: #6190E8;
      font-size: 0.5974rem;
    \\" class=\\"at-segmented-control__item\\">tab1</div><div style=\\"
      color: #6190E8;
      background-color:#fff;
      border-color: #6190E8;
      font-size: 0.5974rem;
    \\" class=\\"at-segmented-control__item\\">tab2</div><div style=\\"
      color: #6190E8;
      background-color:#fff;
      border-color: #6190E8;
      font-size: 0.5974rem;
    \\" class=\\"at-segmented-control__item\\">tab3</div></div>"
`;

exports[`AtSegmentedControl Snap render initial AtSegmentedControl 1`] = `"<div style=\\"border-color: #6190E8;\\" class=\\"at-segmented-control\\"></div>"`;
